crates/full-node/sov-sequencer/src/preferred/cache_warm_up_executor.rs

```diff
@@ -0,0 +1,156 @@
+    }
+
+    pub(crate) fn send_tx(&self, tx: FullyBakedTx) {
+        // Skip update if consumer is too slow.
+        let _ = self.tx_sender.try_send(tx);
+    }
+
+    pub(crate) async fn spawn_execution_task<Rt: Runtime<S>>(
+        info: StateUpdateInfo<S::Storage>,
+        exec_config: RollupBlockExecutorConfig<S>,
+        seq_config: SequencerConfig<S::Address, PreferredSequencerConfig>,
+    ) -> (Self, Vec<JoinHandle<()>>) {
+        let (tx_sender, tx_receiver) = flume::bounded(TX_CHANNEL_SIZE);
+        let (start_block_notification_sender, _) =
+            tokio::sync::broadcast::channel(OPEN_BLOCK_CHANNEL_SIZE);
+
+        let mut handles = Vec::new();
+        for _ in 0..seq_config.sequencer_kind_config.num_cache_warmup_workers {
+            let worker = Self::spawn_worker::<Rt>(
+                info.clone(),
+                exec_config.clone(),
+                seq_config.clone(),
+                tx_receiver.clone(),
+                start_block_notification_sender.subscribe(),
+            );
+
+            handles.push(worker);
+        }
+
+        (
+            Self {
+                tx_sender,
+                start_block_notification_sender,
+            },
+            handles,
+        )
+    }
+
+    fn spawn_worker<Rt: Runtime<S>>(
+        info: StateUpdateInfo<S::Storage>,
+        exec_config: RollupBlockExecutorConfig<S>,
+        seq_config: SequencerConfig<S::Address, PreferredSequencerConfig>,
+        tx_receiver: flume::Receiver<FullyBakedTx>,
+        mut start_block_notification_sender: tokio::sync::broadcast::Receiver<
+            StartBtachNotification<S>,
+        >,
+    ) -> JoinHandle<()> {
+        tokio::spawn(async move {
+            let mut shutdown_receiver = exec_config.shutdown_receiver.clone();
+            let mut executor =
+                RollupBlockExecutor::<_, Rt>::new(&info, exec_config, seq_config.clone());
+
+            let mut is_started = false;
+            loop {
+                tokio::select! {
+                    notify = start_block_notification_sender.recv() => {
+                        match notify{
+                            Ok(notify) => {
+                                let _ = executor.shutdown().await;
+                                Self::start_block(notify, &mut executor).await;
+                                is_started = true;
+                            },
+                            Err(tokio::sync::broadcast::error::RecvError::Lagged(_)) => {
+                                // If the worker is too slow we will just skip some open block updates.
+                                continue;
+                            }
+                            Err(tokio::sync::broadcast::error::RecvError::Closed) => {
+                                // Quit if channel closed.
+                                return
+                            }
+                        }
+                    }
+
+                    tx = tx_receiver.recv_async() => {
```

Comment:
Workers steal transactions from the same queue. Each worker’s transactions will have increasing nonces, but they won’t be strictly sequential.